### PR TITLE
support NHWC in Pool2D, test=develop

### PR DESCRIPTION
**Pull request by zhiqiu**
As the title, Pool2D supports NHWC to achieve better performance.

`paddle.fludi.dygraph.Pool2D` add a keyword argument `data_format`, which indicates the data format of the input and output data. An optional string from: `"NCHW"`, `"NHWC"`. The default is `"NCHW"`

### CODE REVIEW

**Review by phlrain — DISMISSED** [review]

File: python/paddle/fluid/dygraph/nn.py [hunk 1]
```diff
@@ -816,6 +820,11 @@ def __init__(self,
         if not isinstance(use_cudnn, bool):
             raise ValueError("use_cudnn should be True or False")
 
+        if data_format not in ["NCHW", "NHWC"]:
```
Comment (lanxianghit) [inline python/paddle/fluid/dygraph/nn.py:827]: 最好把data_format做一下lowercase或者uppercase处理，兼容各种大小写形式，比如用户输入了nchw，Nchw等

Comment (lanxianghit) [reply python/paddle/fluid/dygraph/nn.py:827]: 上面pool_type等参数类型为str的最好都做一下同样的处理

Comment (zhiqiu) [reply python/paddle/fluid/dygraph/nn.py:827]: 好的，已修改。

**Review by phlrain — DISMISSED** [review]

**Comment by zhiqiu** [comment]
image

image

**Review by lanxianghit — APPROVED** [review]

**Review by swtkiwi — APPROVED** [review]
LGTM

**Pull request merged**